website/docs/cloud-docs/api-docs/workspaces.mdx

```diff
@@ -1097,9 +1103,42 @@ $ curl \
 }
 ```
 
-## Delete a workspace
+## Safe Delete a workspace
+
+Safe deleting a workspace will first check if the workspace has resources under management, and if not it will delete the workspace.
+
+It is considered "unsafe" to delete workspaces with resources under management, as deleting a workspace on Terraform Cloud will leave all infrastructure managed by the workspace orphaned from Terraform.
+
+A workspace can be safe deleted via two endpoints, which behave identically. One refers to a workspace by its ID, and the other by its name and organization.
```

Comment:
Great question! If a run is in progress, then the workspace will be locked (unless someone has force unlocked it), and we consider locked workspaces to have resources under management. So safe delete would fail with an error message explaining that you cant safe delete a locked workspace.

That's actually the reason that we consider locked workspaces to have resources under management.

Do you think it's worth calling out the different criteria for a workspace having RUM here in the API docs? I was leaning towards just trying to do that in one place (like [in this section](https://github.com/hashicorp/terraform-docs-common/pull/146/files#diff-205de5b224e5f306d34c8c4f50b7f7d347d07b7256c74821a9e345375498392eR198) ), but I would be open to putting the RUM criteria in the api docs as well

Reply:
ahhh okay that makes sense! so one thing that is coming very soon (by the end of the month) is a dedicated section of our docs discussing what RUM is, how TFC determines RUM, etc. I think that will be our one central location, and we may even want to update the API docs to link there once it's available. I agree with you that it's better to do it in a centralized location. Since that's coming so soon after this, I think we don't need to expand upon all the criteria here BUT I'd love to have you review the RUM write-up once it's available to make sure it talks through all relevant things (like this) for users.

Reply:
For sure, I am keeping an eye on the development work for the billing and identity team's RUM changes as well (I assume that is other RUM project which will require a docs change).

Now that you mention it, it will be interesting to figure out what we want to refer to as RUM once the B&I teams' changes are ready. We are kinda solving two different problems:

- The B&I team and worried about how many resources under management you have. They don't really care about things like a workspace being locked, or the state still processing
- We (team rocket) just care about whether you might have RUM or not...the actual number of resources doesn't matter too much. But we care a lot about whether there is an ongoing process which might be adding RUM (like a run which is locking the workspace)

So maybe that would be a compelling reason to talk about things like having an ongoing run, or the workspace being locked, etc as reasons that you cant **safe delete** (because the workspace state is in flux), as opposed to being explicitly related to the workspace having RUM.

The current docs changes in this review don't talk about workspace locking having to do with RUM, but it could be a good idea to add the full criteria for when you can/cant safe delete a workspace here in the API, as opposed to in a dedicated RUM section.